### PR TITLE
Suggest evaluating the built-in URLSearchParams first. Fix #37.

### LINKED ISSUE #37: Suggest using the URLSearchParams built-in
The URL standard includes the [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) interface which in many cases should be able to replace this module. @sindresorhus is also [considering](https://github.com/sindresorhus/query-string/issues/307) suggesting it in his `query-string` module README.

This module has 10M weekly downloads. It would really help mentioning the built-in, to help avoid cargo cult usage of this module when not necessary.

The twice as popular [querystring](https://www.npmjs.com/package/querystring) package deprecated itself in favor of URLSearchParams.

### PR DESCRIPTION
**Pull request by tredondo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,10 +2,15 @@
 
 [![Version npm](https://img.shields.io/npm/v/querystringify.svg?style=flat-square)](https://www.npmjs.com/package/querystringify)[![Build Status](https://img.shields.io/github/workflow/status/unshiftio/querystringify/CI/master?label=CI&style=flat-square)](https://github.com/unshiftio/querystringify/actions?query=workflow%3ACI+branch%3Amaster)[![Coverage Status](https://img.shields.io/coveralls/unshiftio/querystringify/master.svg?style=flat-square)](https://coveralls.io/r/unshiftio/querystringify?branch=master)
 
-A somewhat JSON compatible interface for query string parsing. This query string
+First off, see if the built-in [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)
+is suitable for your needs.
+
+Development of this module started in 2014, when `URLSearchParams` wasn't available.
+It provides a somewhat JSON-compatible interface for query string parsing. This query string
```
Comment (jimmywarting) [inline README.md:9]: I would kind of go with node's query string approach. and how the github repo for how querystring [did it](https://github.com/Gozala/querystring)
- in the title, flag it as legacy
- Clearly state it in the beginning that querystringify considered Legacy and new code should use the [<URLSearchParams>](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) API instead. 

When speaking in spec-words and documenting code it helps to be more explicit about types and avoid the word `it` sometimes... as that can have different meaning to different group of peoples.

So I would rephrase the `It provides a somewhat` cuz some might thing `it` refers to `querystringify` and other thing `it` refers to `URLSearchParams`.

also json and objects are **not** the same thing. many ppl over at stackoverflow nit picks on this as well
and also urlencoded is so far of from json that it's not even the same thing. urlencoded don't have numbers, booleans etc... only strings.

I would also avoid linking and perhaps even prompting to yet another query string parsers like sindres

Comment (lpinca) [reply README.md:9]: > I would kind of go with node's query string approach. and how the github repo for how querystring did it

https://github.com/Gozala/querystring is "Node's querystring module for all engines." so it makes sense for them. `querystringify` does not work in the same way. Also "legacy" in does not mean anything. The first sentence here already specifies that alternatives are available.

File: README.md [hunk 2]
```diff
@@ -2,10 +2,15 @@
...
 parser is dumb, don't expect to much from it as it only wants to parse simple
 query strings. If you want to parse complex, multi level and deeply nested
-query strings then you should ask your self. WTF am I doing?
+query strings then you should rethink your approach, due to the [lack of spec
+and numerous edge cases](https://github.com/sindresorhus/query-string#nesting).
```
Comment (lpinca) [inline README.md:13]:
```suggestion
query strings then you should rethink your approach, due to the lack of spec
and numerous edge cases.
```

**Review by lpinca — APPROVED** [review]

**Comment by lpinca** [comment]
Thank you.
